labman/gui/handlers/process_handlers/pooling_process.py

```diff
@@ -416,24 +415,24 @@ def post(self):
         self.write(output)
 
 
-class DownloadPoolFileHandler(BaseHandler):
+class DownloadPoolFileHandler(BaseDownloadHandler):
     @authenticated
     def get(self, process_id):
         try:
             process = PoolingProcess(int(process_id))
         except LabmanUnknownIdError:
             raise HTTPError(404, reason='PoolingProcess %s does not exist'
-                            % process_id)
+                                        % process_id)
         text = process.generate_pool_file()
-
-        filename = 'PoolFile_%s_%s.csv' % (
-            re.sub('[^0-9a-zA-Z\-\_]+', '_',
-                   process.pool.container.external_id), process.id)
-
-        self.set_header('Content-Type', 'text/csv')
-        self.set_header('Expires', '0')
-        self.set_header('Cache-Control', 'no-cache')
-        self.set_header('Content-Disposition', 'attachment; filename='
-                        '%s' % filename)
-        self.write(text)
-        self.finish()
+        plate_names_of_components = [x[0].container.plate.external_id for x in
+                                     process.components]
+
+        plate_names_set = set(plate_names_of_components)
+        if len(plate_names_set) > 1:
+            raise ValueError("Unable to generate normpool file name for pool"
+                             "based on more than one plate: " +
```

Comment:
I think a space is needed before "based"

Reply:
Should ´process.components´ raise on this condition or is it valid in other contexts?

Reply:
Good catch re the space before "based"!  

This error definitely shouldn't be raised on process.components directly, because PoolingProcess objects (what `process` is in this context :) definitely *can* have components from multiple plates--that is exactly what we have when we make amplicon sequencing pools (which are pools of plate pools, and thus draw from multiple plates), and in fact a user COULD even validly chose to make a "plate pool" from more than one amplicon library prep plate (see screenshot).  

<img width="1009" alt="screen shot 2018-11-01 at 1 58 35 pm" src="https://user-images.githubusercontent.com/10677935/47879872-f6ecea80-ddde-11e8-8a51-9f8ca4d2c376.png">

However, the wet lab apparently currently chooses not to do this, and instead chooses to make one pool per library prep plate.  Given this self-imposed limitation, they expect to be able to have the (one-and-only) library plate name embedded in the name of the resulting normpool file.  That *file naming convention* won't work--or at least, won't work as they expect it to--if there are multiple plates represented in the pool, so the error is at the point where the *file name* is generated.  If they decide they want to allow themselves to make plate pools from multiple plates, all they need to do is decide on a more flexible naming convention, and tell us, and we can change the *naming code* and remove this error condition.

I have added the description above into the code as a comment so hopefully it won't be forgotten in the future :)

Reply:
Thank you as well for this explanation, this makes sense.